Feature/init services by priority

TroubleMakerBen opened the pull request:
* I have renamed `MSPriorityHigh` to `MSPriorityMax` to indicate it's the MAX priority. If we need a `MSPriorityHigh` we can introduce this at a later point.
* `MSPriority` has to be ordered, otherwise sorting won't work.
* Modules are sorted based on their priority and then initialized.

jaelim-ms commented:
It shouldn't be coupled with channel priority. Initialize order and channel priority are completely different I think.

clpolet commented on MobileCenter/MobileCenter/MSMobileCenter.m:
```diff
@@ -176,12 +176,36 @@ - (BOOL)configure:(NSString *)appSecret {
+
+    NSArray *sortedServices = [self sortServices:services];
+
+
```
Too much space here :o)

TroubleMakerBen commented:
I have addressed @jaelim-ms 's comment and introduced a new property and added tests for that.
There's also 1 additional check in MSCrashes.mm for an empty data object now that I noticed when I ran our puppet app.

jaelim-ms commented on MobileCenter/MobileCenter/MSMobileCenter.m:
```diff
@@ -176,12 +176,35 @@ - (BOOL)configure:(NSString *)appSecret {
 }
 
+- (NSArray *)sortServices:(NSArray<Class> *)services {
+  // Sort services in descending order to make sure the service with the highest priority gets initialized first.
```
new line. ⚠️

jaelim-ms commented on MobileCenterCrashes/MobileCenterCrashes/MSCrashes.mm:
```diff
@@ -506,17 +510,18 @@ - (void)processLogBufferAfterCrash {
+            [self.logManager processLog:item withPriority:MSPriorityDefault];
+          }
         }
+        // Create empty new file, overwrites the old one.
```
new line
